CommandLine/emake-tests/Parsing/parser-test.cpp

```diff
@@ -315,6 +445,31 @@ TEST(ParserTest, Declarations) {
   EXPECT_EQ(decls->declarations[2].declarator->decl.components.size(), 1);
 }
 
+TEST(ParserTest, Declarations_NoSemicolon) {
+  ParserTester test{"int *x = nullptr, y, (*z)(int x, int) = &y"};
+
+  auto node = test->TryParseStatement();
+  EXPECT_EQ(test->current_token().type, TT_ENDOFCODE);
+  EXPECT_EQ(test.lexer.ReadToken().type, TT_ENDOFCODE);
+
+  ASSERT_EQ(node->type, AST::NodeType::DECLARATION);
+  auto *decls = reinterpret_cast<AST::DeclarationStatement *>(node.get());
+  EXPECT_EQ(decls->def->flags & jdi::DEF_TYPENAME, jdi::DEF_TYPENAME);
+
+  EXPECT_EQ(decls->declarations.size(), 3);
```

Comment:
FYI: Matchers can actually assert that sequences contain a series of other matchers, so if we had one that matched on FullType, you could use
```cpp
ElementsAre(
    FullTypeWith(NoInitiailzer(), DeclarationWithFlags(jdi::DEF_TYPENAME), 
                 RefStackWith(DeclaratorNode::Kind::POINTER_TO)),
    FullTypeWith...)
```

You can get pretty creative with the Matchers interface. Consider it optional, though as you write more of these tests, I think it'll come in handy.